initramfs/init.sh

```diff
@@ -241,19 +270,32 @@ check_and_set_article_number()
             return 0
         fi
 
-        if [ ! -r "${article_number_file}" ]; then
+        if [ ! -r "${article_number_file}" ] && [ ! -r "${country_code_lock_file}" ]; then
             umount "${dev}"
-            echo "No article number file found on '${dev}', skipping."
+            echo "No article number file or country code lock file found on '${dev}', skipping."
             return 0
         fi
 
-        article_number="$(cat "${article_number_file}")"
-        echo "Trying to write article nr: '${article_number}'."
-        # shellcheck disable=SC2086
-        if ! i2ctransfer -y 3 w6@0x57 0x01 0x00 ${article_number}; then
-            umount "${dev}"
-            echo "Failed to write article number to EEPROM, skipping."
-            return 0
+        if [ -r "${article_number_file}" ]; then
+            article_number="$(cat "${article_number_file}")"
+            echo "Trying to write article nr: '${article_number}'."
+            # shellcheck disable=SC2086
+            if ! i2ctransfer -y 3 w6@0x57 0x01 0x00 ${article_number}; then
+                umount "${dev}"
+                echo "Failed to write article number to EEPROM, skipping."
+                return 0
+            fi
+        fi
+
+        if [ -r "${country_code_lock_file}" ]; then
+            country_code="$(cat "${country_code_lock_file}")"
+            echo "Trying to write country code lock: '${country_code}'."
+            # shellcheck disable=SC2086
+            if ! i2ctransfer -y 3 w4@0x57 0x01 0x18 ${country_code}; then
+                umount "${dev}"
+                echo "Failed to write country code lock to EEPROM, skipping."
```

Comment:
Do we need a way to tell ZB that this has failed or will they check the machine before it goes out?

Reply:
They will check the result by looking at the updated "Network" menu in Ultimaker/okuda#871: that will show a successfully locked country.

At this point of system boot, I don't think we have a good way of communicating stuff back to the user...